packages/fast-usdc/src/exos/advancer.js

```diff
@@ -9,116 +11,219 @@ import { addressTools } from '../utils/address.js';
 
 /**
  * @import {HostInterface} from '@agoric/async-flow';
+ * @import {NatAmount} from '@agoric/ertp';
  * @import {ChainAddress, ChainHub, Denom, DenomAmount, OrchestrationAccount} from '@agoric/orchestration';
  * @import {VowTools} from '@agoric/vow';
  * @import {Zone} from '@agoric/zone';
  * @import {CctpTxEvidence, LogFn} from '../types.js';
  * @import {StatusManager} from './status-manager.js';
- * @import {TransactionFeedKit} from './transaction-feed.js';
  */
 
+/**
+ * Expected interface from LiquidityPool
+ *
+ * @typedef {{
+ *   lookupBalance(): NatAmount;
+ *   borrowUnderlying(amount: Amount<"nat">): Promise<PaymentPKeywordRecord>;
+ *   returnUnderlying(principalPayment: Payment<"nat">, feePayment: Payment<"nat">): Promise<void>
+ * }} AssetManagerFacet
+ */
+
+/**
+ * @typedef {{
+ *  chainHub: ChainHub;
+ *  log: LogFn;
+ *  statusManager: StatusManager;
+ *  vowTools: VowTools;
+ * }} AdvancerKitCaps
+ */
+
+/** type guards internal to the AdvancerKit */
+const WatcherHandlersShape = {
```

Comment:
`Shape` is for objects that get passed around. This is an interface guard. It's a kit interface guard but you could still call it `AdvancerI`.

Why is this defined here and then composed with `advancer` in the Exo interface param?